browser/ui/commands/accelerator_service.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_UI_COMMANDS_ACCELERATOR_SERVICE_H_
+#define BRAVE_BROWSER_UI_COMMANDS_ACCELERATOR_SERVICE_H_
+
+#include <string>
+#include <vector>
+
+#include "base/containers/flat_map.h"
+#include "base/observer_list.h"
+#include "base/observer_list_types.h"
+#include "brave/components/commands/browser/accelerator_pref_manager.h"
+#include "brave/components/commands/common/commands.mojom.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+#include "mojo/public/cpp/bindings/pending_remote.h"
+#include "mojo/public/cpp/bindings/receiver_set.h"
+#include "mojo/public/cpp/bindings/remote_set.h"
+#include "ui/base/accelerators/accelerator.h"
+
+class PrefService;
+
+namespace commands {
+
+using Accelerators = base::flat_map<int, std::vector<ui::Accelerator>>;
+
+class AcceleratorService : public mojom::CommandsService, public KeyedService {
+ public:
+  class Observer : public base::CheckedObserver {
+   public:
+    virtual void OnAcceleratorsChanged(const Accelerators& changed) = 0;
+  };
+
+  AcceleratorService(PrefService* pref_service,
+                     const Accelerators& default_accelerators);
+  AcceleratorService(const AcceleratorService&) = delete;
+  AcceleratorService& operator=(const AcceleratorService&) = delete;
+  ~AcceleratorService() override;
```

Comment:
usually a `KeyedService`-based class overrides `Shutdown()` to close/destroy/release things that may trigger some unnecessary events during `~KeyedService()` call (like an access to another Service to remove observer or something like that). Please be aware of this and don't rely on destructor as it may lead to subtle crashes during browser shutdown.

it might be a good thing to add `Shutdown()` override and release `observers_`.